apis/python/src/tiledbsoma/_tiledb_object.py

```diff
@@ -176,6 +176,8 @@ def exists(
                 return md_type.lower() == cls.soma_type.lower()
         except SOMAError:
             return False
+        except tiledb.cc.TileDBError:
```

Comment:
why restrict this to a TileDBError?

I suggest a naked `except`, otherwise we will have the same bug when some other error is raised.

(this is what fsspec and other packages do)

Reply:
> The catch(es) in the implementation should be consolidated into a single naked except statement, to catch any 

@bkmartinjr this is an antipattern -- it would catch things like `NameError` for misspellings, etc.

It's correct to catch only the exceptions appropriate to whether the collection exists or not.

If this is what `fsspec` does then we're propagating an antipattern. I won't die on this hill and am willing to disagree and commit and implement a catch-everything. But I don't (and won't) like it.

Reply:
lolz, no death necessary on _either_ side of the argument.  Let @thetorpedodog  decide :-)

The current implementation catches most of the cases I care about.

Reply:
https://realpython.com/the-most-diabolical-python-antipattern/

Reply:
No disagreement on the theoreticals, but please:
* catch all reasonable exceptions (i.e., OSError, etc) that _shoudl_ return false
* document what it raises so the signature is clear